agent/06_agent_create_cluster.sh

```diff
@@ -72,6 +72,20 @@ function force_mirror_disconnect() {
 
 }
 
+function disable_automated_installation() {
+	local agent_iso_abs_path="$(realpath "${OCP_DIR}/agent.$(uname -p).iso")"
+	local ign_temp_path="$(mktemp --directory)"
+  echo "Extracting ISO ignition..."
+	podman run --privileged --rm -v /run/udev:/run/udev -v "${agent_iso_abs_path}:/data/agent.iso" -w /data  quay.io/coreos/coreos-installer:release iso ignition show agent.iso > "${ign_temp_path}/iso.ign"
+
+  echo "disabling automated installation systemd services..."
+  jq --compact-output --argjson filterlist '["apply-host-config.service", "create-cluster-and-infraenv.service", "set-hostname.service", "start-cluster-installation.service"]' \
+    'walk( . as $i | if type == "object" and has("enabled") and any($filterlist[]; . == $i.name) then .enabled = false else . end)' < "${ign_temp_path}/iso.ign" > "${ign_temp_path}/disabled_automation.ign"
```

Comment:
When I tested this PR, systemctl status does show the four services  are disabled. But create-cluster-and-infraenv.service and set-hostname.service still end up running. 

````
[core@master-0 system]$ systemctl status apply-host-config.service 
● apply-host-config.service - Service that applies host-specific configuration
   Loaded: loaded (/etc/systemd/system/apply-host-config.service; disabled; vendor preset: disabled)
   Active: inactive (dead)

[core@master-0 system]$ systemctl status create-cluster-and-infraenv.service 
● create-cluster-and-infraenv.service - Service that creates initial cluster and infraenv
   Loaded: loaded (/etc/systemd/system/create-cluster-and-infraenv.service; disabled; vendor preset: disabled)
   Active: active (exited) since Wed 2023-01-04 12:14:45 UTC; 8min ago
  Process: 3122 ExecStart=/usr/bin/podman run --net host --cidfile=/run/create-cluster-and-infraenv.service.ctr-id --cgroups=no-conm>
  Process: 3013 ExecStartPre=/usr/local/bin/wait-for-assisted-service.sh (code=exited, status=0/SUCCESS)
  Process: 3011 ExecStartPre=/bin/rm -f /run/create-cluster-and-infraenv.service.ctr-id (code=exited, status=0/SUCCESS)
 Main PID: 3122 (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 100824)
   Memory: 0B
   CGroup: /system.slice/create-cluster-and-infraenv.service

Jan 04 12:14:43 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:43Z" level=info msg="Registering infraenv"
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Registered cluster with id: 3549cfd1-d624-42e6-8e>
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Registering infraenv"
Jan 04 12:14:43 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:43Z" level=info msg="Added 1 nmstateconfigs"
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Added 1 nmstateconfigs"
Jan 04 12:14:45 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:45Z" level=info msg="Registered infraenv with id:>
Jan 04 12:14:45 master-0 podman[3122]: time="2023-01-04T12:14:45Z" level=info msg="Registered infraenv with id: c8d41689-d30b-46fe-a>
Jan 04 12:14:45 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:45Z" level=info msg="Registering extra manifests"
Jan 04 12:14:45 master-0 podman[3122]: time="2023-01-04T12:14:45Z" level=info msg="Registering extra manifests"
Jan 04 12:14:45 master-0 systemd[1]: Started Service that creates initial cluster and infraenv.
...skipping...
● create-cluster-and-infraenv.service - Service that creates initial cluster and infraenv
   Loaded: loaded (/etc/systemd/system/create-cluster-and-infraenv.service; disabled; vendor preset: disabled)
   Active: active (exited) since Wed 2023-01-04 12:14:45 UTC; 8min ago
  Process: 3122 ExecStart=/usr/bin/podman run --net host --cidfile=/run/create-cluster-and-infraenv.service.ctr-id --cgroups=no-conm>
  Process: 3013 ExecStartPre=/usr/local/bin/wait-for-assisted-service.sh (code=exited, status=0/SUCCESS)
  Process: 3011 ExecStartPre=/bin/rm -f /run/create-cluster-and-infraenv.service.ctr-id (code=exited, status=0/SUCCESS)
 Main PID: 3122 (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 100824)
   Memory: 0B
   CGroup: /system.slice/create-cluster-and-infraenv.service

Jan 04 12:14:43 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:43Z" level=info msg="Registering infraenv"
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Registered cluster with id: 3549cfd1-d624-42e6-8e>
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Registering infraenv"
Jan 04 12:14:43 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:43Z" level=info msg="Added 1 nmstateconfigs"
Jan 04 12:14:43 master-0 podman[3122]: time="2023-01-04T12:14:43Z" level=info msg="Added 1 nmstateconfigs"
Jan 04 12:14:45 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:45Z" level=info msg="Registered infraenv with id:>
Jan 04 12:14:45 master-0 podman[3122]: time="2023-01-04T12:14:45Z" level=info msg="Registered infraenv with id: c8d41689-d30b-46fe-a>
Jan 04 12:14:45 master-0 create-cluster-and-infraenv[3134]: time="2023-01-04T12:14:45Z" level=info msg="Registering extra manifests"
Jan 04 12:14:45 master-0 podman[3122]: time="2023-01-04T12:14:45Z" level=info msg="Registering extra manifests"
Jan 04 12:14:45 master-0 systemd[1]: Started Service that creates initial cluster and infraenv.

[core@master-0 system]$ systemctl status set-hostname.service 
● set-hostname.service - Agent-based installer hostname update service
   Loaded: loaded (/etc/systemd/system/set-hostname.service; disabled; vendor preset: disabled)
   Active: active (exited) since Wed 2023-01-04 12:13:14 UTC; 9min ago
  Process: 1427 ExecStart=/usr/local/bin/set-hostname.sh (code=exited, status=0/SUCCESS)
 Main PID: 1427 (code=exited, status=0/SUCCESS)
    Tasks: 0 (limit: 100824)
   Memory: 0B
   CGroup: /system.slice/set-hostname.service

Jan 04 12:13:13 localhost systemd[1]: Starting Agent-based installer hostname update service...
Jan 04 12:13:14 localhost set-hostname.sh[1427]: Host has matching MAC address: 00:be:24:97:3e:63
Jan 04 12:13:14 localhost set-hostname.sh[1427]: Setting hostname to master-0
Jan 04 12:13:14 master-0 set-hostname.sh[1427]: MAC address, 00:be:24:97:3e:67, does not exist on this host
Jan 04 12:13:14 master-0 set-hostname.sh[1427]: MAC address, 00:be:24:97:3e:6b, does not exist on this host
Jan 04 12:13:14 master-0 systemd[1]: Started Agent-based installer hostname update service.

[core@master-0 system]$ systemctl status start-cluster-installation.service 
● start-cluster-installation.service - Service that starts cluster installation
   Loaded: loaded (/etc/systemd/system/start-cluster-installation.service; disabled; vendor preset: disabled)
   Active: inactive (dead)
````

Reply:
set-hostname probably gets pulled as a dependency of agent.service

Reply:
create-cluster-and-infraenv is Wanted by install-status.service, so that's probably what is starting it. That's a recent change too, so it probably wasn't happening when Toni tested this PR.

Reply:
Thanks. Added it to the filter list.

Reply:
set-hostname.service starts up. Should we add agent.service to the filter list?